### PR TITLE
Add new functions for junos modules (lock, unlock, load)

### PR DESCRIPTION
**Pull request by rajvidhimar**
### What does this PR do?
* Add new functions - lock, load and unlock
* Update install_config to lock database before installing config.
* Minor changes in documentation

### What issues does this PR fix or reference?
NA

### Previous Behavior
install_config did not lock configuration before loading the new one.

### New Behavior
Now, install_config locks the configuration first.

### Tests written?
No

### CODE REVIEW

File: salt/states/junos.py [hunk 1]
```diff
@@ -281,7 +288,7 @@ def install_config(name, **kwargs):
 
     Parameters:
       Required
-        * path or template_path (Either one of the keyword must be there):
+        * name:
```
Comment (vnitinv) [inline salt/states/junos.py:291]: https://github.com/Juniper/salt/pull/26/files#diff-442286d51776ddefc791bc7f15e42fc2R272
&
https://github.com/Juniper/salt/pull/26/files#diff-442286d51776ddefc791bc7f15e42fc2R282

uses path and template_path, then we have removed this description and replaced it with name

**Review by vnitinv — COMMENTED** [review]
check my comment

File: salt/states/junos.py (hunk 1 above)
Comment (rajvidhimar) [reply salt/states/junos.py:291]: We are using the name only..the earlier documentation was not correct.

File: salt/modules/junos.py [hunk 2]
```diff
@@ -1079,3 +1097,186 @@ def file_copy(src=None, dest=None):
...
+
+
+def lock():
+    """
+    Attempts an exclusive lock on the candidate configuration. This
+    is a non-blocking call.
+
+    .. note::
+        Any user who wishes to use lock, must necessarily unlock the
+        configuration too.
```
Comment (stacywsmith) [inline salt/modules/junos.py]: Not only must the user call unlock, they should make sure they call unlock in the same orchestration run in which the lock was called.

**Review by stacywsmith — CHANGES_REQUESTED** [review]
Everything you have looks good, but don't we also need to implement a commit_check() method in order to implement the workflow we've been discussing?

**Review by vnitinv — APPROVED** [review]

**Pull request merged**